File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;
map<char, int> arr;
string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    vector<char> tmp={'R','T','C','F','J','M','A','N'};
    for(int i=0;i<8;i++){
        arr.insert(make_pair(tmp[i],0));
    }
    
     
    
    for(int i=0;i<choices.size();i++){
        if(choices[i]<4){
            arr[survey[i][0]]=arr[survey[i][0]]+(4-choices[i]);
        }
        else if(choices[i]==4){
            continue;
        }
        else{
            arr[survey[i][1]]=arr[survey[i][1]]+(choices[i]-4);
        }
            
    }
    
    for(auto em: arr){
        cout<<em.first<<" "<<em.second<<endl;
    }
    
   
    
    arr['R']>=arr['T'] ? answer+='R' : answer+='T';
    arr['C']>=arr['F'] ? answer+='C' : answer+='F';
    arr['J']>=arr['M'] ? answer+='J' : answer+='M';
    arr['A']>=arr['N'] ? answer+='A' : answer+='N';
    
    
    
    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        HashMap<Character, Integer> map = new HashMap<>();
        map.put('R',0);
        map.put('T',0);
        map.put('C',0);
        map.put('F',0);
        map.put('J',0);
        map.put('M',0);
        map.put('A',0);
        map.put('N',0);
        String answer = "";
        for(int i=0;i<choices.length;i++){
            if(choices[i]==4) continue;
            else{
                if(choices[i]==1){
                    int val = map.get(survey[i].charAt(0));
                    map.put(survey[i].charAt(0), val+=3);
                }
                else if(choices[i]==2){
                    int val = map.get(survey[i].charAt(0));
                    map.put(survey[i].charAt(0), val+=2);
                }
                else if(choices[i]==3){
                    int val = map.get(survey[i].charAt(0));
                    map.put(survey[i].charAt(0), val+=1);
                }
                else if(choices[i]==5){
                    int val = map.get(survey[i].charAt(1));
                    map.put(survey[i].charAt(1), val+=1);
                }
                else if(choices[i]==6){
                    int val = map.get(survey[i].charAt(1));
                    map.put(survey[i].charAt(1), val+=2);
                }
                else if(choices[i]==7){
                    int val = map.get(survey[i].charAt(1));
                    map.put(survey[i].charAt(1), val+=3);
                }
            }
        }
        String[] arr = {"RT","CF","JM","AN"};
        for(int i=0;i<4;i++){
            char val1 = arr[i].charAt(0);
            char val2 = arr[i].charAt(1);
            int num1 = map.get(val1);
            int num2 = map.get(val2);
            if(num1>num2){
                answer+=val1;
            }
            else if(num1<num2){
                answer+=val2;
            }
            else{
                if((int)val1>(int)val2){
                    answer+=val2;
                }
                else answer+=val1;
            }
        }
        return answer;
    }
}